backend/pkg/api/applications.go

```diff
@@ -2,25 +2,47 @@ package api
 
 import (
 	"database/sql"
+	"fmt"
+	"regexp"
+	"strings"
+	"sync"
 	"time"
 
 	"github.com/doug-martin/goqu/v9"
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
 	"gopkg.in/guregu/null.v4"
 )
 
+type appsCache map[string]string
+
+var (
+	// cachedApps caches the mapping of apps' product-ids and UUIDs to apps' UUIDs.
+	// The UUID -> UUID seeming redundancy is because this way we can use it to
+	// validate also the UUIDs against existing apps.
+	// It must not be modified directly but replaced (atomically or via lock)
+	// by a new map to prevent data races.
+	// An update must be triggered through updateCachedApps() each time
+	// an apps change. A RW lock was chosen to prevent data
+	// races over the pointer itself.
+	cachedApps     appsCache
```

Comment:
cachedApps gives the perception that apps are cached but only the mapping of productID and appID is.

```suggestion
	cachedProductIDAppID     map[string][string]
```

Reply:
I'll go with `cachedAppIDs` because Map is the type and that's not too relevant as well as it can lead to more changes if we ever change it.